scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -415,7 +415,7 @@ class Router(formatOps: FormatOps) {
         val caseStat = leftOwner.asInstanceOf[Case]
         if (isCaseBodyABlock(tok, caseStat)) Seq(Split(Space, 0))
         else {
-          def newlineSplit(cost: Int) = {
+          def newlineSplit(cost: Int)(implicit line: sourcecode.Line) = {
```

Comment:
Does `sourcecode.Line` somehow affects formatting?

Or is this PR just for unit test addition?

Reply:
it absolutely does. the ordering of State uses the line of the associated Split, so this line affects the position in the BestFirstSearch priority queue.

Reply:
and let me mention once again that i normally add a test in a separate commit, before modifying the code in which the test also changes, so you could review each commit separately and see for yourself.

Reply:
added explanation in the commit message and the PR description

Reply:
got it, thank you for explanation